tests/test_models.py

```diff
@@ -468,3 +471,29 @@ def test_can_save_wsn_device(self):
 		self.assertIsNotNone(device.pk)
 		self.assertIsNotNone(device.date_created)
 		self.assertEqual(device.date_created.date(), timezone.now().date())
+
+	def test_gcm_is_default_cloud_message_type_when_not_specified_in_settings(self) -> None:
+		self._validate_device_cloud_message_type(expected_cloud_message_type="GCM")
+
+	@override_settings()
+	def test_cloud_message_type_is_set_to_gcm(self) -> None:
+		settings.PUSH_NOTIFICATIONS_SETTINGS.update({
+			"DEFAULT_CLOUD_MESSAGE_TYPE": "GCM",
+		})
+		self._validate_device_cloud_message_type(expected_cloud_message_type="GCM")
+
+	@override_settings()
+	def test_cloud_message_type_is_set_to_fcm(self) -> None:
+		settings.PUSH_NOTIFICATIONS_SETTINGS.update({
+			"DEFAULT_CLOUD_MESSAGE_TYPE": "FCM",
+		})
+		self._validate_device_cloud_message_type(expected_cloud_message_type="FCM")
+
+	def _validate_device_cloud_message_type(self, expected_cloud_message_type: str) -> None:
+		# Reload the models so that cached model is evicted and
+		# field default value is re-read from settings
+		import push_notifications.models
+		importlib.reload(push_notifications.models)
+		from push_notifications.models import GCMDevice
+		field_object = GCMDevice._meta.get_field("cloud_message_type")
+		self.assertEqual(field_object.default, expected_cloud_message_type)
```

Comment:
all of this should become obsolete and can be remove after switching to the callable default.

Reply:
instead I would recommend to create model instances in each of the tests and assert they have been created with the correct `cloud_message_type`